DataStructureMemories/DataStructureMemories/DataStructureViewControllers/DetailViewController.swift

```diff
@@ -0,0 +1,137 @@
+        view.layoutIfNeeded()
+    }
+    
+    func giveWayToWKWebViewController() {
+        let mainStoryboard = UIStoryboard(name: "Main", bundle: nil)
+        guard let navigationWKController = mainStoryboard.instantiateViewController(withIdentifier: "showWK") as? WKWebNavigationViewController else { return }
+        navigationWKController.exactURL = srtuctureCell?.getWikiLink()
+        self.present(navigationWKController, animated: true)
+    }
+    func giveWayToUIWebViewController() {
+        let mainStoryboard = UIStoryboard(name: "Main", bundle: nil)
+
+        guard let uiWebController = mainStoryboard.instantiateViewController(withIdentifier: "UIWebViewController") as? UIWebViewController else { return }
+            
+        uiWebController.exactURL = srtuctureCell?.getWikiLink()
+        self.present(uiWebController, animated: true)
+    }
+    func giveWayToSFSafaryViewController() {
+        let mainStoryboard = UIStoryboard(name: "Main", bundle: nil)
+        guard let sfSafaryViewController = mainStoryboard.instantiateViewController(withIdentifier: "SFSafaryViewController") as? SFSafaryViewController
+            else { return }
+        sfSafaryViewController.exactURL = srtuctureCell?.getWikiLink()
+        self.present(sfSafaryViewController, animated: true)
+    }
+    
+    @IBAction func pathToLinksToWiki(_ sender: Any) {
+    
+        let actionSheetController: UIAlertController = UIAlertController( title: "Please select", message: "Option to select", preferredStyle: .actionSheet)
+        let cancelActionButton = UIAlertAction(title: "Cancel", style: .cancel) {
+            _ in
+            print("Cancel")
+        }
+        actionSheetController.addAction(cancelActionButton)
+        
+        let uiWebViewAction = UIAlertAction(title: "UIWebViewAction", style: .default) {
+            _ in
+            print("Save")
+            self.giveWayToUIWebViewController()
+        }
+        actionSheetController.addAction(uiWebViewAction)
+        
+        let webKitView = UIAlertAction(title: "WebViewAction", style: .default) {
+            _ in
+            self.giveWayToWKWebViewController()
+            print("webKitView")
+        }
+        actionSheetController.addAction(webKitView)
+        
+        let sfSafary = UIAlertAction(title: "SfSafaryAction", style: .default) {
+            _ in
+             self.giveWayToSFSafaryViewController()
+            print("sfSafary")
+        }
+        actionSheetController.addAction(sfSafary)
+
+        self.present(actionSheetController, animated: true, completion: nil)
+    }
+    
+    @IBAction func buttonVisualize(_ sender: UIButton) {
+        let mainStoryboard = UIStoryboard( name: "Main", bundle: nil )
+        guard let visualizeViewController = mainStoryboard.instantiateViewController(withIdentifier: "VisualizationViewController") as? VisualizationViewController else { return }
+        guard let title = srtuctureCell?.titleOfString().lowercased(),
```

Comment:
I would suggest to add `var type: ATDTypes` into `srtuctureCell` to avoid re-calculation